### DISCUSSION TITLE
Recommended Java version for agent extensions

### DISCUSSION THREAD

**Discussion by EddeCCC** [open-telemetry/opentelemetry-java-instrumentation#10769] [Q&A] [answered]
Greetings,

since I couldn't find a clear answer to this, I would like to ask you here.
Is there any recommended or mandatory Java source/target version for agent extensions?

The [example extensions](https://github.com/open-telemetry/opentelemetry-java-instrumentation/tree/main/examples/extension) use `compileJava { options.release.set(8) }` in their build.gradle. This prevents us to use modern language features (like enhanced switch expressions), but not the use of "modern" dependency versions (like Spring Boot 3).

For example, we would like to include Spring Boot 3, which only supports JDK17+, into our extension.
Could this create challenges with your Java agent? 
We are planning to use our agent extension only for applications with at least JDK 17.

**Comment by laurit** [comment, marked as answer]
You can compile your extension with whatever version works for you. In this project we usually choose the version used for compiling instrumentation based on the version of instrumented library. If the library works with jdk 8 we also compile our instrumentation for jdk 8 to ensure that the instrumentation works when the library works.
